modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -3228,12 +3229,17 @@ private Integer findVarcharColumnLength(AttributeDescriptor att) {
             if (sqlTypeName != null) {
                 sqlTypeNames[i] = sqlTypeName;
             }
-        }
+        }        
+        //Step 2 Geos-7533 make below two lines controllable from hint or param
+        //if all sql type names have been found in dialect dont
+        //go to database GEOS-7533
+        boolean allTypesFound = !ArrayUtils.contains(sqlTypeNames, null);        
+        if(allTypesFound) return sqlTypeNames;
```

Comment:
Space between `ìf`` and ``(`` missing, remember to do a full build to apply automatic formatting.

Reply:
I would add a log messages (level warning) if the type doesn't exist and we need to go to the database.

Reply:
I will add the warning message and push changes 👍

Reply:
fixed